### PR TITLE
[QRadar v3] Add Resilience

### PR DESCRIPTION
**Pull request by DeanArbel**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/45541

## Description
Add graceful error handling where applicable.

### CODE REVIEW

File: Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py [hunk 1]
```diff
@@ -1919,6 +1949,7 @@ def qradar_offense_update_command(client: Client, args: Dict) -> CommandResults:
         )
 
     if closing_reason_name:
+        # if this call fails raise an error and stop command execution
```
Comment (tkatzir) [inline Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py:1952]:
```suggestion
        # if this call fails, raise an error and stop command execution
```

Comment (tkatzir) [reply Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py:1952]: Appears numerous times...

**Review by tkatzir — CHANGES_REQUESTED** [review]

File: Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py [hunk 2]
```diff
@@ -2507,6 +2553,7 @@ def qradar_reference_set_value_upsert_command(client: Client, args: Dict) -> Com
     if date_value:
         values = [get_time_parameter(value, epoch_format=True) for value in values]
 
+    # if one of these calls fail raise an error and stop command execution
```
Comment (tkatzir) [inline Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py:2556]:
```suggestion
    # if one of these calls fail, raise an error and stop command execution
```

**Review by tkatzir — CHANGES_REQUESTED** [review]

**Review by tkatzir — APPROVED** [review]

File: Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py [hunk 3]
```diff
@@ -821,12 +821,16 @@ def get_offense_types(client: Client, offenses: List[Dict]) -> Dict:
...
-    return {offense_type.get('id'): offense_type.get('name') for offense_type in offense_types}
+    try:
+        offense_types_ids = {offense.get('offense_type') for offense in offenses if offense.get('offense_type') is not None}
+        if not offense_types_ids:
+            return dict()
+        offense_types = client.offense_types(filter_=f'''id in ({','.join(map(str, offense_types_ids))})''',
+                                             fields='id,name')
+        return {offense_type.get('id'): offense_type.get('name') for offense_type in offense_types}
+    except Exception as e:
+        demisto.info(f"Encountered an issue while getting offense type: {e}")
```
Comment (anara123) [inline Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py:832]: @DeanArbel why not demisto.error?

Comment (tkatzir) [reply Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py:832]: > @DeanArbel why not demisto.error?

We wanted something more than `debug`.
`error` is indeed more suitable here.

**Pull request merged**